purldb-toolkit/README.rst

```diff
@@ -20,84 +25,541 @@ Use this command to get basic help::
 
     $ purlcli --help
     Usage: purlcli [OPTIONS] COMMAND [ARGS]...
-    
+
       Return information from a PURL.
-    
+
     Options:
       --help  Show this message and exit.
-    
+
     Commands:
       metadata  Given one or more PURLs, for each PURL, return a mapping of...
```

Comment:
Here instead of three dots at the end, we can show the full texts which are there when we run `purlcli --help`, for all 4 commands here. Otherwise the truncated texts are not that useful.

Reply:
Actually @AyanSinhaMahapatra when I ran `purlcli --help` I got the same truncated texts.  I'll postpone this change until you've had a chance to read this reply in case that changes your thinking.  I like the idea of more info, but as you'll see in the examples below, the description for instance for `metadata` is rather long -- just want to make sure we're OK with the way this wraps due to its length when we insert it into the `purlcli --help` output block in the RTD.

This is `purlcli --help`:

```
$ purlcli --help
Usage: purlcli [OPTIONS] COMMAND [ARGS]...

  Return information from a PURL.

Options:
  --help  Show this message and exit.

Commands:
  metadata  Given one or more PURLs, for each PURL, return a mapping of...
  urls      Given one or more PURLs, for each PURL, return a list of all...
  validate  Check the syntax and upstream repo status of one or more PURLs.
  versions  Given one or more PURLs, return a list of all known versions...
```

And this is `purlcli metadata --help`:

```
$ purlcli metadata --help
Usage: purlcli metadata [OPTIONS]

  Given one or more PURLs, for each PURL, return a mapping of metadata fetched
  from the fetchcode package.py info() function.

Options:
  --purl TEXT        PackageURL or PURL.
  --output FILENAME  Write meta output as JSON to FILE.  [required]
  --file FILENAME    Read a list of PURLs from a FILE, one per line.
  --help             Show this message and exit.
```